python/pyarrow/src/arrow/python/helpers.cc

```diff
@@ -22,6 +22,7 @@
 
 #include <cmath>
 #include <limits>
+#include <mutex>
```

Comment:
@pitrou Should we conditionally include this only `#ifdef Py_GIL_DISABLED`?

Reply:
We could, but I don't think think it's important.